tests/testthat/test-req-perform-stream.R

```diff
@@ -4,3 +4,43 @@ test_that("req_stream() is deprecated", {
     resp <- req_stream(req, identity, buffer_kb = 32)
   )
 })
+
+test_that("can buffer to lines", {
+  lines <- character()
+  accumulate_lines <- function(x) {
+    lines <<- c(lines, strsplit(rawToChar(x), "\n")[[1]])
+    TRUE
+  }
+
+  # Each line is 225 bytes, should should be split into ~2 pieces
+  resp <- request_test("/stream/10") |>
+    req_perform_stream(accumulate_lines, buffer_kb = 0.1, round = "line")
+  expect_equal(length(lines), 10)
+
+  valid_json <- map_lgl(lines, jsonlite::validate)
+  expect_equal(valid_json, rep(TRUE, 10))
+})
+
+test_that("eventually terminates even if never rounded", {
+  out <- raw()
+  accumulate <- function(x) {
+    out <<- c(out, x)
+    TRUE
+  }
+
+  resp <- request_test("/stream-bytes/1024") |>
+    req_perform_stream(
+      accumulate,
+      buffer_kb = 0.1,
+      round = function(bytes) integer()
+    )
+  expect_equal(length(out), 0)
```

Comment:
Is this correct, or should it return 1024? i.e. should we always call `callback` on the final bytes in the buffer?

Reply:
I think `0L` makes sense here. Should the response be different though in that case ? Is this a 408 ?

Reply:
Well the HTTP request succeeded, and returned all it's data, but it never used a cut point we expected. I'm now leaning more towards 0 being wrong, because it's surprising for data to silent vanish if there's not a trailing newline (which must be fairly common since `readLines()` handles it).

Reply:
😱 yeah, tweaked it the `incomplete` logic a bit to accommodate for these cases.